NXDOC-2367: add documentation for nuxeo-coldstorage addon (#1867)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-add-ons/nuxeo-coldstorage/nuxeo-coldstorage-functional-overview.md

---
title: Functional Overview
description: Discover Nuxeo coldstorage capabilities.
review:
    comment: ''
    date: ''
    status: ''
labels:
    - coldstorage
    - glacier
    - storage
toc: true
tree_item_index: 100
---

> ## Coldstorage

Reviewer: @jaubenque can you confirm if Coldstorage should be used as a single word or 2 (Cold Storage) please?

Author: I confirmed with Julien it should be Cold Storage and corrected all the instances

> The nuxeo coldstorage (glacier) connector is a long term storage solution that allows Nuxeo clients to take advantage of cost savings by using AWS Cold Storage for large files and/or less frequently accessed content. 

Reviewer: @jaubenque can you confirm the official name of the connector please?

## Send to Coldstorage
Once the nuxeo rendition of the document is available, the user can trigger the flow by clicking on "send to coldstorage" button.

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Send to Coldstorage button
    name: SendToCS1.png
    addins#screenshot#up_to_date
--}}

> This will trigger an AWS workflow that will change the class to "glacier" class. You can check the details at the [AWS documentation here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-transition-general-considerations.html).

Author: yes great point

A confirmation toast will be briefly displayed and a banner will be visible on the document stating that the document is processed by nuxeo coldstorage.

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Send to Coldstorage Confirmation
    name: SendToCS2.png
    addins#screenshot#up_to_date
--}}
![Send to Coldstorage Confirmation](nx_asset://2970838e-c5d6-4438-a0cf-eb2fcf4671f9)

## Send multiple documents to Coldstorage

> There's an option to send multiple documents to coldstorage, if all of them satisfy the requirements (they are not yet in coldstorage) and if the user has the wrights. Select the files that you want to upload and there will be an action button to move them to coldstorage.

Reviewer 2: ```suggestion
There's an option to send multiple documents to coldstorage, if all of them satisfy the requirements (they are not yet in coldstorage) and if the user has the rights. Select the files that you want to upload and there will be an action button to move them to coldstorage.
```

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Send Multiple To Coldstorage
    name: SendMultipleToCS1.png
    addins#screenshot#up_to_date
--}}
![Send Multiple To Coldstorage](nx_asset://d6f239bd-b830-4b3e-a041-bc23ec2d985c)

## Retrieve from Coldstorage

> When the user wants to have temporary acess to the full file for download, he can trigger the retrieve flow. It is required that the file is already fully stored in nuxeo coldstorage - in other terms, that the document AWS storage class transitioned to glacier.

Reviewer 2: ```suggestion
When the user wants to have temporary access to the full file for download, she can trigger the retrieve flow. It is required that the file is already fully stored in nuxeo coldstorage - in other terms, that the document AWS storage class transitioned to glacier.
```

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Restore from coldstorage button
    name: RestoreFromCS1.png
    addins#screenshot#up_to_date
--}}
![Restore from coldstorage button](nx_asset://bd6ceed7-004f-4d78-a0e7-2740b4871ce0)

> The user can then trigger the flow by clicking on the restore button. As an asyncronous process, it is expected that it takes some time. An email is sent to the user when the retrieve process is done and also the document will display a banner stating that the retrieve is in process.

Reviewer 2: ```suggestion
The user can then trigger the flow by clicking on the restore button. As an asynchronous process, it is expected that it takes some time. An email is sent to the user when the retrieve process is done and also the document will display a banner stating that the retrieve is in process.
```

> The user can then trigger the flow by clicking on the restore button. As an asyncronous process, it is expected that it takes some time. An email is sent to the user when the retrieve process is done and also the document will display a banner stating that the retrieve is in process.

Reviewer 3: The user can then trigger the flow by clicking on the **retrieve** button

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Retrieve from Coldstorage
    name: RetrieveFromCS2.png
    addins#screenshot#up_to_date
--}}
![Retrieve from Coldstorage](nx_asset://d606f54c-7113-4ce6-a319-be5486c40159)

Once the retrieve is done, the banner will display information about the preview availability and the option to download the content.

{{!--     ### nx_asset ###
    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/Nuxeo Coldstorage/Retrieved from Coldstorage 
    name: RetrieveFromCS3.png
    addins#screenshot#up_to_date
--}}
![Retrieved from Coldstorage ](nx_asset://514f7467-bce2-404c-9d76-0864c1869b57)

> By default, the file will be available for download for 24h. For now, that value can be changed for all files by adding a configuration on nuxeo-conf. So, for example, if we want the file 

Reviewer 2: ```suggestion
By default, the file will be available for download for 24h. For now, that value can be changed for all files by adding a configuration on the `nuxeo.conf` file. So, for example, if we want the file 
```

> By default, the file will be available for download for 24h. For now, that value can be changed for all files by adding a configuration on nuxeo-conf. So, for example, if we want the file 

Reviewer 2: ```suggestion
By default, the file will be available for download for 24h. For now, that value can be changed for all files by adding a configuration to the `nuxeo.conf` file. So, for example, if we want the file 
```

Reviewer 2: "So, for example, if we want the file..." what to what? It seems that something is missing here.

Author: yes hehe

```

> nuxeo.coldstorage.numberOfDaysOfAvailability.value.default=

Reviewer 2: We should specify what the unit here is. Perhaps even give an example. If I want it to be available for 48h, what should we specify? 48 or 2?

> nuxeo.coldstorage.numberOfDaysOfAvailability.value.default= 2

Reviewer 2: Could you please update that on https://nuxeowiki.atlassian.net/wiki/spaces/NT/pages/2036072489/Nuxeo+Cold+Storage+Glacier+Connector as well? Thank you.

```

## Restore from Coldstorage

> When the user doesn't want the file to be stored in Coldstorage anymore, he can request a restore by clicking on the restore button. The operation is also asynchronous, and the user will get notified via email once it's completed and the banner will dissapear. The send to coldstorage button will be available again.

Reviewer 2: ```suggestion
When the user doesn't want the file to be stored in Coldstorage anymore, she can request a restore by clicking on the restore button. The operation is also asynchronous, and the user will get notified via email once it's completed and the banner will disappear. The send to Coldstorage button will be available again.
```
